File: akka-benchmark/src/main/java/io/sarl/akka/hierarchy/Child.java
package io.sarl.akka.hierarchy;

import akka.actor.AbstractActor;
import akka.actor.Props;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class Child extends AbstractActor {
    static public Props props() {
        return Props.create(Child.class, () -> new Child());
    }

    static public class PrintSomething {
        public final String something;

        public PrintSomething(String something) {
            this.something = something;
        }
    }

    private LoggingAdapter log = Logging.getLogger(getContext().getSystem(), this);

    public Child() {
        //
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .match(PrintSomething.class, event -> {
                    log.info(event.something);
                })
                .build();
    }
}

File: akka-benchmark/src/main/kotlin/io/sarl/akka/hierarchy/Child.kt
package io.sarl.akka.hierarchy

import akka.actor.AbstractActor
import akka.actor.Props
import akka.event.Logging
import akka.event.LoggingAdapter

class Child : AbstractActor() {

    private val log = Logging.getLogger(context.system, this)

    class PrintSomething(val something: String)

    override fun createReceive(): AbstractActor.Receive {
        return receiveBuilder()
                .match(PrintSomething::class.java) { event -> log.info(event.something) }
                .build()
    }

    companion object {
        fun props(): Props {
            return Props.create(Child::class.java) { Child() }
        }
    }
}//
